### PR TITLE
Refactor resource quota polling logic

### LINKED ISSUE #4902: Refactor resource quota polling logic
**What would you like to cleanup?**:
#### Current behaviour
Resource quotas are polled every time we navigate to a page where project quota widget is visible, therefore there is a delay until the widget is visible
#### Required refactoring
The resource quota's data should be saved once retrieved from the API, so it would show instantly once loaded.
Once the resource quota widget is visible, we should still keep updating them from API.

### PR DESCRIPTION
**Pull request by Talha-Jamil-TJ**
**What this PR does / why we need it**:

A delay was observed in the resource quota widget while navigating between different routes, this PR fixes it.

**Which issue(s) this PR fixes**:

Fixes #4902 

**What type of PR is this?**
/kind cleanup

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: src/app/dynamic/enterprise/quotas/service.ts [hunk 1]
```diff
@@ -70,20 +75,25 @@ export class QuotaService {
...
+            this._previousQuotaMap.set(projectId, quota);
+          }
+        })
       );
 
       this.quotaMap.set(projectId, quota$);
     }
 
-    return this.quotaMap.get(projectId);
+    return this.quotaMap.get(projectId).pipe(startWith(this._previousQuotaMap.get(projectId) ?? null));
```
Comment (Waseem826) [inline src/app/dynamic/enterprise/quotas/service.ts:96]: Are we handling this `null` value where we call this method?

Comment (Talha-Jamil-TJ) [reply src/app/dynamic/enterprise/quotas/service.ts:96]: Yes, I've put in `filter( Boolean)` where ever I'm. subscribing to quotas
image

Comment (Talha-Jamil-TJ) [reply src/app/dynamic/enterprise/quotas/service.ts:96]: Yes, isLoading is being set to false instantly, I'm catering for this.

**Review by Waseem826 — APPROVED** [review]
/approve

**Pull request merged**